test/resource/http.js

```diff
@@ -165,9 +173,92 @@ exports.request = function _request(path, request) {
   setImmediate(() => {
     emitter.emit('request', req, res);
 
-    req.write(request.BODY);
+    if (request.BODY) {
+      req.write(request.BODY);
+    }
+
     req.end();
   });
 
   return emitter;
 };
+
+/**
+ * Wraps stub request and response instances and a promise for testing
+ * HTTP Server request handlers
+ */
+class Bench {
+  /**
+   * @constructor
+   * @param {Object}  request A set of parameters for the request
+   * @param {Function}  handle  The HTTP request handler
+   */
+  constructor(request, handle) {
+    this.promise = new Promise((resolve) => {
+      const req = new IncomingMessage(request);
+      const res = new ServerResponse(() => resolve([req, res]));
+
+      setImmediate(() => {
+        handle(req, res);
+
+        if (request.BODY) { req.write(request.BODY); }
+        req.end();
+      });
+    });
+  }
+
+  /**
+   * Attach a post-response handler to the test bench for assertions
+   * @param {Function}  handle  Accepts standard `(request, response)` handler arguments
+   * @return  {Bench} self
+   */
+  response(handle) {
+    this.promise = this.promise.then((args) => handle(args[0], args[1]));
+
+    return this;
+  }
+
+  /**
+   * Add resolved/rejected handlers to the underlying promise
+   * @param {Function}  resolved  Receives an array `[request, response]` as the success argument
+   * @param {Function}  rejected  Error handler
+   * @return  {Bench} self
+   */
+  then(resolved, rejected) {
+    this.promise = this.promise.then(resolved, rejected);
+
+    return this;
+  }
+
+  /**
+   * Add a rejected handler to the underlying promise
+   * @param {Function}  rejected  Error handler
+   * @return  {Bench} self
+   */
+  catch(rejected) {
+    this.promise = this.promise.catch(rejected);
+
+    return this;
+  }
+
+  /**
+   * Attach the same handler to resolve and reject
+   *
+   * This is a helper for `next([err])` interfaces, where a callback
+   * must be called at the end of the routine, with an optional argument
+   * indicating that an error occurred.
+   *
+   * @param {Function}  handle
+   * @return  {Bench} self
+   */
+  finally(handle) {
+    this.then(() => handle());
+    this.catch((err) => handle(err));
+
+    return this;
+  }
+}
+
+exports.bench = function(request, handler) {
```

Comment:
Adding a `"func-names": 0` rule to [test/.eslintrc.json](https://github.com/rapid7/turnstile/blob/v0.1.1/test/.eslintrc.json) should get the linter to not care about this exported test function.